text/3-entrees/fixed-point.md

```diff
@@ -0,0 +1,327 @@
+
+When programmers learn to use non-integer numbers in their programs, they are usually taught to use [floating point](https://en.wikipedia.org/wiki/Floating-point_arithmetic)s. In blockchain, we use an alternative representation of fractional numbers called [fixed point](https://en.wikipedia.org/wiki/Fixed-point_arithmetic). There are several ways to use fixed point numbers, and this recipe will introduce three of them. In particular we'll see:
+
+* Substrate's own fixed point structs and traits
+* The [substrate-fixed](https://github.com/encointer/substrate-fixed/) library
+* A manual fixed point implementation (and why it's nicer to use a library)
+* A comparison of the two libraries in a compounding interest example
+
+## What's Wrong with Floats?
+
+Floats are cool for all kinds of reasons, but they also have one important drawback. Floating point arithmetic is **nondeterministic**. In order for the nodes in a blockchain network to reach agreement on the state of the chain, all operations must be completely deterministic. Luckily fixed point arithmetic is deterministic, and is often not much harder to use once you get the hang of it.
+
+## Multiplicative Accumulators
+*[`pallets/fixed-point`](https://github.com/substrate-developer-hub/recipes/tree/master/pallets/fixed-point)*
+
+The first pallet covered in this recipe contains three implementations of a multiplicative accumulator. That's a fancy way to say the pallet lets users submit fractional numbers and keeps track of the product from multiplying them all together. The value starts out at one (the [multiplicative identity](https://en.wikipedia.org/wiki/Identity_element)), and it gets multiplied by whatever values the users submit. These three independent implementations compare and contrast the features of each.
+
+### Permill Accumulator
+
+We'll using the most common approach which takes it's fixed point implementation from Substrate itself. There are a few fixed-point structs available in Substrate, all of which implement the [`PerThing` trait](https://substrate.dev/rustdocs/master/sp_arithmetic/trait.PerThing.html), that cover different amounts of precision. For this accumulator example, we'll use the [`PerMill` struct](https://substrate.dev/rustdocs/master/sp_arithmetic/struct.Permill.html) which represents fractions as parts per million. There are also [`Perbill`](https://substrate.dev/rustdocs/master/sp_arithmetic/struct.Perbill.html), [`PerCent`](https://substrate.dev/rustdocs/master/sp_arithmetic/struct.Percent.html), and [`PerU16`](https://substrate.dev/rustdocs/master/sp_arithmetic/struct.PerU16.html), which all provide the same interface (because it comes from the trait). Substrate's fixed-point structs are somewhat unique because they represent _only_ fractional parts of numbers. That means they can represent numbers between 0 and 1 inclusive, but _not_ numbers with whole parts like 2.718 or 3.14.
```

Comment:
Possessive "its" has no apostrophe.

```suggestion
We'll using the most common approach which takes its fixed point implementation from Substrate itself. There are a few fixed-point structs available in Substrate, all of which implement the [`PerThing` trait](https://substrate.dev/rustdocs/master/sp_arithmetic/trait.PerThing.html), that cover different amounts of precision. For this accumulator example, we'll use the [`PerMill` struct](https://substrate.dev/rustdocs/master/sp_arithmetic/struct.Permill.html) which represents fractions as parts per million. There are also [`Perbill`](https://substrate.dev/rustdocs/master/sp_arithmetic/struct.Perbill.html), [`PerCent`](https://substrate.dev/rustdocs/master/sp_arithmetic/struct.Percent.html), and [`PerU16`](https://substrate.dev/rustdocs/master/sp_arithmetic/struct.PerU16.html), which all provide the same interface (because it comes from the trait). Substrate's fixed-point structs are somewhat unique because they represent _only_ fractional parts of numbers. That means they can represent numbers between 0 and 1 inclusive, but _not_ numbers with whole parts like 2.718 or 3.14.
```